CMake refactored + small changes

kabicm opened the pull request:
This PR brings the following changes:
- **CMake:** fully refactored, added installation option.
- **Macros defined:**  macros for entering/leaving profiling regions added.
- **Profiler reset:** previously, `profiler_clear()` function was not implemented, although recorder class implements this functionality. Now this function is also implemented. This is especially useful when multiple runs of the library to be profiled are performed and we want to exclude the "warm-up" run from the profiler data.

Commit:
Using target_include_directories instead of the global include.

teonnik committed:
Refactor CMake

- Provide support for installation

teonnik committed:
Use GNUInstallDirs

teonnik committed:
Add archive destination

teonnik committed:
Remove obsolete subproject guards

teonnik committed:
Make installation optional

Commit:
Added clear() function to the profiler.

teonnik committed:
Guard target

teonnik committed:
Merge branch 'master' of github.com:kabicm/semiprof

bcumming commented on examples/omp.cpp:
```diff
@@ -3,7 +3,7 @@
 #include <omp.h>
 
-#include <semiprof/semiprof.hpp>
+#include <semiprof.hpp>
```
Keep the old version, and make the CMakeLists.txt so that the include path is `include`, not `include/semiprof`

kabicm replied:
Thanks, should be changed now.

bcumming reviewed (changes requested):
Thanks! One small comment/change.

Commit:
Changed include path.

Commit:
Changing include paths in CMake.

Commit:
Changing include paths in CMake.

teonnik committed:
Fix bug with installing when semiprof is a subproject

- When semiprof can be found as a subprojec multiple times in the
  dependecy tree of a project, an error occurs due to duplicate install
  commands. This commit prevents the error by guarding installation by
  checking whether the list of targets is empty. The list is only
  non-empty the first time the semiprof target is defined. This works
  because targets are global.

kabicm commented:
Hi Ben, how is this PR looking to you now?

Commit:
Merge remote-tracking branch 'upstream/master'
Updating the fork with the original.

kabicm commented:
This PR can be closed, since a newer [PR](https://github.com/bcumming/semiprof/pull/3) is opened.
